src/org/armedbear/lisp/SocketStream.java

```diff
@@ -47,6 +47,10 @@ public SocketStream(Socket socket, Stream in, Stream out)
         this.socket = socket;
     }
 
+    public Socket getJavaSocket() {
```

Comment:
I don't have a problem making this interface public, but please separate out into a different pull request.